lib/TH/generic/THStorage.c

```diff
@@ -107,9 +107,9 @@ void THStorage_(free)(THStorage *storage)
   if(!storage)
     return;
 
-  if((storage->flag & TH_STORAGE_REFCOUNTED) && (storage->refcount > 0))
+  if((storage->flag & TH_STORAGE_REFCOUNTED) && (THAtomicGet(&storage->refcount) > 0))
   {
```

Comment:
Euh, I wrote this in a hurry. Basically:
- Before we did not have TH_STORAGE_REFCOUNTED, and the way to not refcount a tensor/storage was to put the count at a negative value. The second part is thus to support this old behavior.
- The first part of the line is to support the new behavior.

I could remove the second part, as it is unlikely people are using this way now.